### ISSUE TITLE
Cannot upload files to Vagrant environment

### ISSUE THREAD

**Issue by pdurbin** [IQSS/dataverse#3745]
Tested on 9e8e766 after a bug report at https://groups.google.com/d/msg/dataverse-community/nFGSWblzuPI/KDo6YpQNAwAJ from Michiel Meulendijk, it is impossible to upload files to Dataverse from within the Vagrant environment mentioned at http://guides.dataverse.org/en/4.6.1/installation/prep.html#vagrant-for-testing-only

The screen flashes and then the file disappears. Here's a screenshot after the file disappears:

![screen shot 2017-04-04 at 8 58 36 am](https://cloud.githubusercontent.com/assets/21006/24657717/5fc50834-1915-11e7-9273-6d56c4a0258b.png)

Here's the stacktrace:

```
[2017-04-04T05:57:45.307-0700] [glassfish 4.1] [WARNING] [] [javax.enterprise.resource.webcontainer.jsf.lifecycle] [tid: _ThreadID=49 _ThreadName=jk-connector(1)] [timeMillis: 1491310665307] [levelValue: 900] [[
  /editFilesFragment.xhtml @84,65 fileUploadListener="#{EditDatafilesPage.handleFileUpload}": java.lang.NoClassDefFoundError: edu/harvard/hul/ois/jhove/viewer/ConfigWindow
javax.el.ELException: /editFilesFragment.xhtml @84,65 fileUploadListener="#{EditDatafilesPage.handleFileUpload}": java.lang.NoClassDefFoundError: edu/harvard/hul/ois/jhove/viewer/ConfigWindow
	at com.sun.faces.facelets.el.TagMethodExpression.invoke(TagMethodExpression.java:111)
	at org.primefaces.component.fileupload.FileUpload.broadcast(FileUpload.java:318)
	at javax.faces.component.UIViewRoot.broadcastEvents(UIViewRoot.java:755)
	at javax.faces.component.UIViewRoot.processDecodes(UIViewRoot.java:931)
	at com.sun.faces.lifecycle.ApplyRequestValuesPhase.execute(ApplyRequestValuesPhase.java:78)
	at com.sun.faces.lifecycle.Phase.doPhase(Phase.java:101)
	at com.sun.faces.lifecycle.LifecycleImpl.execute(LifecycleImpl.java:198)
	at javax.faces.webapp.FacesServlet.service(FacesServlet.java:646)
	at org.apache.catalina.core.StandardWrapper.service(StandardWrapper.java:1682)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:344)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:214)
	at org.glassfish.tyrus.servlet.TyrusServletFilter.doFilter(TyrusServletFilter.java:295)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:256)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:214)
	at org.ocpsoft.rewrite.servlet.RewriteFilter.doFilter(RewriteFilter.java:205)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:256)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:214)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:316)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:160)
	at org.apache.catalina.core.StandardPipeline.doInvoke(StandardPipeline.java:734)
	at org.apache.catalina.core.StandardPipeline.invoke(StandardPipeline.java:673)
	at com.sun.enterprise.web.WebPipeline.invoke(WebPipeline.java:99)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:174)
	at org.apache.catalina.connector.CoyoteAdapter.doService(CoyoteAdapter.java:415)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:282)
	at com.sun.enterprise.v3.services.impl.ContainerMapper$HttpHandlerCallable.call(ContainerMapper.java:459)
	at com.sun.enterprise.v3.services.impl.ContainerMapper.service(ContainerMapper.java:167)
	at org.glassfish.grizzly.http.server.HttpHandler.runService(HttpHandler.java:201)
	at org.glassfish.grizzly.http.server.HttpHandler.doHandle(HttpHandler.java:175)
	at org.glassfish.grizzly.http.server.HttpServerFilter.handleRead(HttpServerFilter.java:235)
	at org.glassfish.grizzly.filterchain.ExecutorResolver$9.execute(ExecutorResolver.java:119)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:284)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:201)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.execute(DefaultFilterChain.java:133)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.process(DefaultFilterChain.java:112)
	at org.glassfish.grizzly.ProcessorExecutor.execute(ProcessorExecutor.java:77)
	at org.glassfish.grizzly.nio.transport.TCPNIOTransport.fireIOEvent(TCPNIOTransport.java:561)
	at org.glassfish.grizzly.strategies.AbstractIOStrategy.fireIOEvent(AbstractIOStrategy.java:112)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.run0(WorkerThreadIOStrategy.java:117)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.access$100(WorkerThreadIOStrategy.java:56)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy$WorkerThreadRunnable.run(WorkerThreadIOStrategy.java:137)
	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.doWork(AbstractThreadPool.java:565)
	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.run(AbstractThreadPool.java:545)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.NoClassDefFoundError: edu/harvard/hul/ois/jhove/viewer/ConfigWindow
	at edu.harvard.hul.ois.jhove.DefaultConfigurationBuilder.writeDefaultConfigFile(Unknown Source)
	at edu.harvard.hul.ois.jhove.JhoveBase.init(Unknown Source)
	at edu.harvard.iq.dataverse.util.JhoveFileType.checkFileType(JhoveFileType.java:86)
	at edu.harvard.iq.dataverse.util.JhoveFileType.getFileMimeType(JhoveFileType.java:186)
	at edu.harvard.iq.dataverse.util.FileUtil.determineFileType(FileUtil.java:306)
	at edu.harvard.iq.dataverse.util.FileUtil.createDataFiles(FileUtil.java:599)
	at edu.harvard.iq.dataverse.EditDatafilesPage.handleFileUpload(EditDatafilesPage.java:1790)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.sun.el.parser.AstValue.invoke(AstValue.java:289)
	at com.sun.el.MethodExpressionImpl.invoke(MethodExpressionImpl.java:304)
	at org.jboss.weld.util.el.ForwardingMethodExpression.invoke(ForwardingMethodExpression.java:40)
	at org.jboss.weld.el.WeldMethodExpression.invoke(WeldMethodExpression.java:50)
	at com.sun.faces.facelets.el.TagMethodExpression.invoke(TagMethodExpression.java:105)
	... 43 more
Caused by: java.lang.ClassNotFoundException: edu.harvard.hul.ois.jhove.viewer.ConfigWindow
	at org.glassfish.web.loader.WebappClassLoader.loadClass(WebappClassLoader.java:1783)
	at org.glassfish.web.loader.WebappClassLoader.loadClass(WebappClassLoader.java:1633)
	... 59 more
]]

[2017-04-04T05:57:45.311-0700] [glassfish 4.1] [SEVERE] [] [javax.enterprise.resource.webcontainer.jsf.context] [tid: _ThreadID=49 _ThreadName=jk-connector(1)] [timeMillis: 1491310665311] [levelValue: 1000] [[
  java.lang.NoClassDefFoundError: edu/harvard/hul/ois/jhove/viewer/ConfigWindow
	at edu.harvard.hul.ois.jhove.DefaultConfigurationBuilder.writeDefaultConfigFile(Unknown Source)
	at edu.harvard.hul.ois.jhove.JhoveBase.init(Unknown Source)
	at edu.harvard.iq.dataverse.util.JhoveFileType.checkFileType(JhoveFileType.java:86)
	at edu.harvard.iq.dataverse.util.JhoveFileType.getFileMimeType(JhoveFileType.java:186)
	at edu.harvard.iq.dataverse.util.FileUtil.determineFileType(FileUtil.java:306)
	at edu.harvard.iq.dataverse.util.FileUtil.createDataFiles(FileUtil.java:599)
	at edu.harvard.iq.dataverse.EditDatafilesPage.handleFileUpload(EditDatafilesPage.java:1790)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.sun.el.parser.AstValue.invoke(AstValue.java:289)
	at com.sun.el.MethodExpressionImpl.invoke(MethodExpressionImpl.java:304)
	at org.jboss.weld.util.el.ForwardingMethodExpression.invoke(ForwardingMethodExpression.java:40)
	at org.jboss.weld.el.WeldMethodExpression.invoke(WeldMethodExpression.java:50)
	at com.sun.faces.facelets.el.TagMethodExpression.invoke(TagMethodExpression.java:105)
	at org.primefaces.component.fileupload.FileUpload.broadcast(FileUpload.java:318)
	at javax.faces.component.UIViewRoot.broadcastEvents(UIViewRoot.java:755)
	at javax.faces.component.UIViewRoot.processDecodes(UIViewRoot.java:931)
	at com.sun.faces.lifecycle.ApplyRequestValuesPhase.execute(ApplyRequestValuesPhase.java:78)
	at com.sun.faces.lifecycle.Phase.doPhase(Phase.java:101)
	at com.sun.faces.lifecycle.LifecycleImpl.execute(LifecycleImpl.java:198)
	at javax.faces.webapp.FacesServlet.service(FacesServlet.java:646)
	at org.apache.catalina.core.StandardWrapper.service(StandardWrapper.java:1682)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:344)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:214)
	at org.glassfish.tyrus.servlet.TyrusServletFilter.doFilter(TyrusServletFilter.java:295)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:256)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:214)
	at org.ocpsoft.rewrite.servlet.RewriteFilter.doFilter(RewriteFilter.java:205)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:256)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:214)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:316)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:160)
	at org.apache.catalina.core.StandardPipeline.doInvoke(StandardPipeline.java:734)
	at org.apache.catalina.core.StandardPipeline.invoke(StandardPipeline.java:673)
	at com.sun.enterprise.web.WebPipeline.invoke(WebPipeline.java:99)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:174)
	at org.apache.catalina.connector.CoyoteAdapter.doService(CoyoteAdapter.java:415)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:282)
	at com.sun.enterprise.v3.services.impl.ContainerMapper$HttpHandlerCallable.call(ContainerMapper.java:459)
	at com.sun.enterprise.v3.services.impl.ContainerMapper.service(ContainerMapper.java:167)
	at org.glassfish.grizzly.http.server.HttpHandler.runService(HttpHandler.java:201)
	at org.glassfish.grizzly.http.server.HttpHandler.doHandle(HttpHandler.java:175)
	at org.glassfish.grizzly.http.server.HttpServerFilter.handleRead(HttpServerFilter.java:235)
	at org.glassfish.grizzly.filterchain.ExecutorResolver$9.execute(ExecutorResolver.java:119)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:284)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:201)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.execute(DefaultFilterChain.java:133)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.process(DefaultFilterChain.java:112)
	at org.glassfish.grizzly.ProcessorExecutor.execute(ProcessorExecutor.java:77)
	at org.glassfish.grizzly.nio.transport.TCPNIOTransport.fireIOEvent(TCPNIOTransport.java:561)
	at org.glassfish.grizzly.strategies.AbstractIOStrategy.fireIOEvent(AbstractIOStrategy.java:112)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.run0(WorkerThreadIOStrategy.java:117)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.access$100(WorkerThreadIOStrategy.java:56)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy$WorkerThreadRunnable.run(WorkerThreadIOStrategy.java:137)
	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.doWork(AbstractThreadPool.java:565)
	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.run(AbstractThreadPool.java:545)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.ClassNotFoundException: edu.harvard.hul.ois.jhove.viewer.ConfigWindow
	at org.glassfish.web.loader.WebappClassLoader.loadClass(WebappClassLoader.java:1783)
	at org.glassfish.web.loader.WebappClassLoader.loadClass(WebappClassLoader.java:1633)
	... 59 more
]]
```

**Comment by pdurbin** [comment]
I just made pull request #3746 and dragged it to Code Review at https://waffle.io/IQSS/dataverse

**Comment by pdurbin** [comment]
"Coverage increased (+0.4%) to 10.117%" 😄 

Also, I've been discussing this with Don and Michiel at http://irclog.iq.harvard.edu/dataverse/2017-04-04

**Comment by pdurbin** [comment]
Here's a screenshot of file upload working in Vagrant in the 3745-vagrant-cannot-upload-files branch (the one behind pull request #3746). The key to the fix/workaround is that /home/glassfish/glassfish4/glassfish/domains/domain1/config/jhove.conf was not present ( 0d41e31 ). As @donsizemore points out, this could also be fixed in the installer ("it's the sed line 1050 in the install script" -- http://irclog.iq.harvard.edu/dataverse/2017-04-04#i_51037). While I was in there I added a JUnit test for FileUtil.determineFileType (and improved the code). There were no tests in this area which is why code coverage is increasing a sizable percent (+0.4%).

**Comment by pdurbin** [comment]
Whoops, the screenshot of it working in pull request #3746:

![screen shot 2017-04-04 at 10 36 30 am](https://cloud.githubusercontent.com/assets/21006/24662549/56b5c2ca-1923-11e7-8bdc-5b18ef1e0af2.png)

**Comment by pdurbin** [comment]
Ok @landreev put a better fix in place at 8a4443f which allowed me to pull out the workaround I added ( 71f69fb ). I'm putting this in QA at https://waffle.io/IQSS/dataverse
